lib/gather/gather-telemetry.js

```diff
@@ -28,235 +29,24 @@ module.exports = async function gatherTelemetry(url, puppeteerArgs = {}) {
-        'Helper',
-        'Router',
-        'Engine',
-      ];
-      return types.find(type => Ember[type] && object instanceof Ember[type]) || 'EmberObject';
-    }
-
-    /**
-     * Parses ember meta data object and collects the runtime information
-     *
-     * @param {Object} meta
-     * @returns {Object} data - Parsed metadata for the ember object
-     * @returns {String[]} data.computedProperties - list of computed properties
-     * @returns {String[]} data.getters - list of ES5 getters
-     * @returns {String[]} data.observedProperties - list of observed properties
-     * @returns {Object} data.observerProperties - list of observer properties
-     * @returns {Object} data.offProperties - list of observer properties
-     * @returns {String[]} data.overriddenActions - list of overridden actions
-     * @returns {String[]} data.overriddenProperties - list of overridden properties
-     * @returns {String[]} data.ownProperties - list of object's own properties
-     * @returns {String} data.type - type of ember object
-     * @returns {Object} data.unobservedProperties - list of unobserved properties
-     */
-    function parseMeta(meta = {}) {
-      if (!meta || !meta.source) {
-        return {};
-      }
-      const { source } = meta;
-      const type = getType(source);
-
-      const ownProperties = Object.keys(source).filter(key => !['_super', 'actions'].includes(key));
+          if (module && module.default && module.default.proto) {
+            let defaultProto = module.default.proto();
 
-      const ownActions = source.actions ? Object.keys(source.actions) : [];
-
-      const observedProperties = Object.keys(meta._watching || {});
-
-      const overriddenProperties = ownProperties.filter(key => isOverridden(meta.parent, key));
-
-      const overriddenActions = ownActions.filter(key => isActionOverridden(meta.parent, key));
-
-      const computedProperties = [];
-      meta.forEachDescriptors((name, desc) => {
-        const descProto = Object.getPrototypeOf(desc) || {};
-        const constructorName = descProto.constructor ? descProto.constructor.name : '';
-        if (
-          desc.enumerable &&
-          ownProperties.includes(name) &&
-          constructorName === 'ComputedProperty'
-        ) {
-          computedProperties.push(name);
-        }
-      });
-
-      const ownDesc = Object.getOwnPropertyDescriptors(source);
-      const getters = Object.keys(ownDesc).filter(
-        key => ownDesc[key].get && !computedProperties.includes(key)
-      );
-
-      const { offProperties, unobservedProperties } = ownProperties.reduce(
-        ({ offProperties, unobservedProperties }, key) => {
-          const { type, events } = getListenerData(meta.parent, key);
-          if (type === 'event') {
-            offProperties[key] = events;
-          } else if (type === 'observer') {
-            unobservedProperties[key] = events;
+            telemetry[modulePath] = window.gatherFn(defaultProto);
```

Comment:
@rwjblue should we keep this return logic? Just want to double check

Reply:
yeah, the idea here would be that the main thing this library does is:

* loop over all modules known
* call the provided "gather function"
* collect the return value of the gather function for each module
* provide the cumulative results to the codemod